rust/ql/src/queries/security/CWE-825/AccessInvalidPointer.qhelp

```diff
@@ -0,0 +1,49 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+<overview>
+
+<p>
+Dereferencing an invalid or dangling pointer is undefined behavior. Memory may be corrupted
+causing the program to crash or behave incorrectly, in some cases exposing the program to
+potential attacks.
+</p>
+
+</overview>
+<recommendation>
+
+<p>
+When dereferencing a pointer in <code>unsafe</code> code, take care that the pointer is valid and
+points to the intended data. Code may need to be rearranged or additional checks added to ensure
+safety in all circumstances. If possible, rewrite the code using safe Rust types to avoid this
+class of problems altogether.
```

Comment:
Not sure my comment is valid here. Are you using `class` as in a class used in software development? If not, I suggest we replace with `kind` or `sort`.
```suggestion
kind of problems altogether.
```

Reply:
Yep, good point.